test/shared-class.test.js

```diff
@@ -243,6 +268,26 @@ describe('SharedClass', function() {
       expect(methods).to.not.contain(DYN_METHOD_NAME);
     });
   });
+
+  describe('sharedClass.disableMethodByName(methodName)', function() {
+    it('excludes disabled static methods from the method list', function() {
+      var METHOD_NAME = 'testMethod';
+      var sc = new SharedClass('SomeClass', SomeClass);
+      var sm = sc.defineMethod(METHOD_NAME, {isStatic: true});
+      sc.disableMethodByName(METHOD_NAME);
+      var methods = sc.methods().map(getName);
+      expect(methods).to.not.contain(METHOD_NAME);
+    });
+
+    it('excludes disabled prototype methods from the method list', function() {
+      var INST_METHOD_NAME = 'prototype.instTestMethod';
+      var sc = new SharedClass('SomeClass', SomeClass);
+      var sm = sc.defineMethod(INST_METHOD_NAME, {isStatic: false});
```

Comment:
This is incorrect API usage which creates the situation when you need to add checks whether method name contains `prototype.` prefix.

This is the correct way for registering a new prototype method:

```js
sc.defineMethod('instTestMethod', { isStatic: false });
```